In [1]:
# importing the necessary libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [4]:
data = pd.read_csv("household_power_consumption.txt",sep=';')
data

C:\Users\bhattade\AppData\Local\Temp\ipykernel_13532\3715934069.py:1: DtypeWarning: Columns (2,3,4,5,6,7) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv("household_power_consumption.txt",sep=';')


Date      Time Global_active_power Global_reactive_power  \
0        16/12/2006  17:24:00               4.216                 0.418   
1        16/12/2006  17:25:00               5.360                 0.436   
2        16/12/2006  17:26:00               5.374                 0.498   
3        16/12/2006  17:27:00               5.388                 0.502   
4        16/12/2006  17:28:00               3.666                 0.528   
...             ...       ...                 ...                   ...   
2075254  26/11/2010  20:58:00               0.946                   0.0   
2075255  26/11/2010  20:59:00               0.944                   0.0   
2075256  26/11/2010  21:00:00               0.938                   0.0   
2075257  26/11/2010  21:01:00               0.934                   0.0   
2075258  26/11/2010  21:02:00               0.932                   0.0   

         Voltage Global_intensity Sub_metering_1 Sub_metering_2  \
0        234.840           18.400          0.000          1.000   
1        233.630           23.000          0.000          1.000   
2        233.290           23.000          0.000          2.000   
3        233.740           23.000          0.000          1.000   
4        235.680           15.800          0.000          1.000   
...          ...              ...            ...            ...   
2075254   240.43              4.0            0.0            0.0   
2075255    240.0              4.0            0.0            0.0   
2075256   239.82              3.8            0.0            0.0   
2075257    239.7              3.8            0.0            0.0   
2075258   239.55              3.8            0.0            0.0   

         Sub_metering_3  
0                  17.0  
1                  16.0  
2                  17.0  
3                  17.0  
4                  17.0  
...                 ...  
2075254             0.0  
2075255             0.0  
2075256             0.0  
2075257             0.0  
2075258             0.0  

[2075259 rows x 9 columns]

In [7]:
data_sample = data.sample(50000) # the original data has 20 lakhs records. let's work on 50000 records for this task.

In [8]:
data_sample

Date      Time Global_active_power Global_reactive_power  \
1426281    2/9/2009  04:45:00               0.308                 0.136   
1633096   23/1/2010  19:40:00               4.118                 0.294   
672962    28/3/2008  01:26:00               0.400                 0.228   
52570     22/1/2007  05:34:00               0.214                 0.000   
1789565   12/5/2010  11:29:00               1.574                 0.308   
...             ...       ...                 ...                   ...   
645447     8/3/2008  22:51:00               4.342                 0.194   
1397139   12/8/2009  23:03:00               1.730                 0.286   
678110    31/3/2008  15:14:00               0.332                 0.130   
1376985   29/7/2009  23:09:00               1.232                 0.000   
1578265  16/12/2009  17:49:00               0.788                 0.232   

         Voltage Global_intensity Sub_metering_1 Sub_metering_2  \
1426281  243.150            1.400          0.000          2.000   
1633096  235.200           18.000         31.000          1.000   
672962   241.940            1.800          0.000          1.000   
52570    243.490            0.800          0.000          0.000   
1789565  241.410            6.600          0.000          1.000   
...          ...              ...            ...            ...   
645447   239.430           18.000         38.000          1.000   
1397139  240.150            7.200          0.000          1.000   
678110   246.690            1.400          0.000          0.000   
1376985  240.780            5.000          0.000          0.000   
1578265  248.710            3.200          0.000          1.000   

         Sub_metering_3  
1426281             1.0  
1633096            18.0  
672962              0.0  
52570               0.0  
1789565            19.0  
...                 ...  
645447              0.0  
1397139            18.0  
678110              0.0  
1376985            19.0  
1578265             0.0  

[50000 rows x 9 columns]

In [10]:
data_sample.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 50000 entries, 1426281 to 1578265
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Date                   50000 non-null  object 
 1   Time                   50000 non-null  object 
 2   Global_active_power    50000 non-null  object 
 3   Global_reactive_power  50000 non-null  object 
 4   Voltage                50000 non-null  object 
 5   Global_intensity       50000 non-null  object 
 6   Sub_metering_1         50000 non-null  object 
 7   Sub_metering_2         50000 non-null  object 
 8   Sub_metering_3         49370 non-null  float64
dtypes: float64(1), object(8)
memory usage: 3.8+ MB


In [11]:
data_sample.columns

Index(['Date', 'Time', 'Global_active_power', 'Global_reactive_power',
       'Voltage', 'Global_intensity', 'Sub_metering_1', 'Sub_metering_2',
       'Sub_metering_3'],
      dtype='object')

In [14]:
data_sample.isnull().sum()

Date                       0
Time                       0
Global_active_power        0
Global_reactive_power      0
Voltage                    0
Global_intensity           0
Sub_metering_1             0
Sub_metering_2             0
Sub_metering_3           630
dtype: int64

We have 630 null values in Sub_metering_3 feature